MdeModulePkg/Bus/Pci/PciBusDxe/PciEnumeratorSupport.c

```diff
@@ -222,16 +222,21 @@ PciSearchDevice (
 {
   PCI_IO_DEVICE  *PciIoDevice;
   BOOLEAN        IgnoreOptionRom;
+  UINT32         Seg = 0;
+  EFI_PCI_ROOT_BRIDGE_IO_PROTOCOL              *PciRootBridgeIo;
 
   PciIoDevice     = NULL;
   IgnoreOptionRom = FALSE;
+  PciRootBridgeIo = Bridge->PciRootBridgeIo;
+  Seg = PciRootBridgeIo->SegmentNumber;
 
   DEBUG ((
     DEBUG_INFO,
-    "PciBus: Discovered %s @ [%02x|%02x|%02x]  [VID = 0x%x, DID = 0x%0x]\n",
+    "PciBus: Discovered %s @ [%04x|%02x|%02x|%02x]  [VID = 0x%x, DID = 0x%0x]\n",

```

Comment:
Can you check if all the below codes should be updated to include Segment information?

Edk2 • MdeModulePkg\Bus\Pci\PciBusDxe\PciCommand.c:
  238          DEBUG_WARN,
  239:         "%a: [%02x|%02x|%02x] failed to access config space at offset 0x%x\n",
  240          __func__,

Edk2 • MdeModulePkg\Bus\Pci\PciBusDxe\PciEnumerator.c:
  995          DEBUG_ERROR,
  996:         "PciBus: [%02x|%02x|%02x] was rejected due to resource confliction.\n",
  997          PciResNode->PciDev->BusNumber,

Edk2 • MdeModulePkg\Bus\Pci\PciBusDxe\PciEnumeratorSupport.c:
  230      DEBUG_INFO,
  231:     "PciBus: Discovered %s @ [%02x|%02x|%02x]  [VID = 0x%x, DID = 0x%0x]\n",
  232      IS_PCI_BRIDGE (Pci) ?     L"PPB" :

Edk2 • MdeModulePkg\Bus\Pci\PciBusDxe\PciLib.c:
  233            DEBUG_INFO,
  234:           "   Base = 0x%lx;\tLength = 0x%lx;\tAlignment = 0x%lx;\tOwner = %s [%02x|%02x|%02x:",
  235            Bar[Resource->Bar].BaseAddress,

  347        DEBUG_INFO,
  348:       "Bridge [%02x|%02x|%02x]\n",
  349        Bridge->BusNumber,

  428            DEBUG_ERROR,
  429:           "PciBus: [%02x|%02x|%02x] Adjust Pci Device Bar Size\n",
  430            PciIoDevice->BusNumber,